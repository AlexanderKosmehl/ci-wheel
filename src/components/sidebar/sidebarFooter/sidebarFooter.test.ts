import {
  describe, expect, it, vi,
} from 'vitest';
import generateSidebarFooter from './sidebarFooter';

describe('generateSidebarFooter', () => {
  const onImportClick = vi.fn(() => {});
  const onArchiveClick = vi.fn(() => {});

  const footer = generateSidebarFooter({
    importOnClick: onImportClick,
    archiveOnClick: onArchiveClick,
  });

  const [importButton, archiveButton] = footer.querySelectorAll<HTMLButtonElement>('button');

  it('generates component correctly', () => {
    expect(footer).toMatchSnapshot();
  });

  it('adds working callbacks', () => {
    if (!importButton || !archiveButton) throw Error('No import or archive button rendered!');

    importButton.click();
    expect(onImportClick).toBeCalled();

    archiveButton.click();
    expect(onArchiveClick).toBeCalled();
  });
});
