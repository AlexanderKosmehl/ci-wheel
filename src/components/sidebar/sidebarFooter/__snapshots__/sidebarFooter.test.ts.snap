// Vitest Snapshot v1

exports[`generateSidebarFooter > generates component correctly 1`] = `
<div
  class="container"
>
  <button
    class="footerButton"
  >
    Import
  </button>
  <button
    class="footerButton"
  >
    Archiv
  </button>
</div>
`;
